tests/ResultQueryTest.php

```diff
@@ -588,4 +588,164 @@ public function testQueryStreamFromLazyConnectionWillErrorWhenConnectionIsClosed
 
         $connection->close();
     }
+
+    protected function checkMaxAllowedPacket($connection, $min = 0x1100000)
+    {
+        return $connection->query('SHOW VARIABLES LIKE \'max_allowed_packet\'')->then(
+            function ($res) use ($min, $connection) {
+                $current = $res->resultRows[0]['Value'];
+                if ($current < $min) {
+                    throw new \Exception('max_allowed_packet too low: current: ' . $current . ' min: ' . $min);
+                }
+                return \React\Promise\resolve();
+            }
+        )->then(
+            function () {
+                return true;
+            }
+        );
+    }
+
+    /**
+     * This should not trigger splitted packets sending
+     */
+    public function testSelectStaticTextSplitPacketsExactlyBelow16MiB()
+    {
+        $connection = $this->createConnection(Loop::get());
+
+        $promise = $this->checkMaxAllowedPacket($connection, 0x1000000); // 16MiB
+
+        $promise->then(
+            function () use ($connection) {
+                /**
+                 * This should be exactly below 16MiB packet
+                 *
+                 * x03 + "select ''" = len(10)
+                 */
+                $text = str_repeat('A', 0xffffff - 11);
+                $connection->query('select \'' . $text . '\'')->then(function (QueryResult $command) use ($text) {
+                    $this->assertCount(1, $command->resultRows);
+                    $this->assertCount(1, $command->resultRows[0]);
+                    $this->assertSame($text, reset($command->resultRows[0]));
+                })->done();
+            }
+        )->otherwise(
+            function (\Throwable $e) {
+                $this->markTestIncomplete('checkMaxAllowedPacket: ' . $e->getMessage());
+            }
+        )->always(
+            function () use ($connection) {
+                $connection->quit();
+            }
+        )->done();
+
+        Loop::run();
+    }
+
+    /**
+     * This should trigger split packets sending and
+     * will send additional empty packet to signal to the server that split packets has ended.
+     */
+    public function testSelectStaticTextSplitPacketsExactly16MiB()
+    {
+        $connection = $this->createConnection(Loop::get());
+
+        $promise = $this->checkMaxAllowedPacket($connection);
+
```

Comment:
I don't think this test behaves the way you intended to. As I see it, this test will be marked as skipped if someone uses a `max_allowed_packet` size of 16 MiB which should be allowed in this case. This is caused by the `$min` default of your `checkMaxAllowedPacket()` function, as it will set the minimum to 17MiB and throws an exception for configurations with 16 MiB.